pyproject.toml

```diff
@@ -0,0 +1,61 @@
+[project]
+name = 'pyshtools'
+dynamic = [
+  'version',
+]
+description = 'SHTOOLS - Spherical Harmonic Tools'
+readme = 'README.md'
+license = { text = 'BSD-3-Clause' }
+requires-python = '>=3.6'
+keywords = ['Spherical Harmonics', 'Spectral Estimation', 'Slepian Functions',
+            'Legendre Functions', 'Gravity Field', 'Magnetic Field']
+classifiers = [
+  'Development Status :: 5 - Production/Stable',
+  'Environment :: Console',
+  'Intended Audience :: Science/Research',
+  'Intended Audience :: Developers',
+  'License :: OSI Approved :: BSD License',
+  'Natural Language :: English',
+  'Operating System :: OS Independent',
+  'Programming Language :: Fortran',
+  'Programming Language :: Python',
+  'Programming Language :: Python :: 3',
+  'Topic :: Scientific/Engineering',
+  'Topic :: Scientific/Engineering :: GIS',
+  'Topic :: Scientific/Engineering :: Mathematics',
+  'Topic :: Scientific/Engineering :: Physics',
+]
+authors = [
+  { name = 'The SHTOOLS developers', email = 'mark.wieczorek@ipgp.fr' },
+]
+dependencies = [
+  'numpy',
```

Comment:
is meson-python a dependency here?

Reply:
Also, I am not 100% sure if we should add the following or not:

pkg-config: required by meson-python I think.
jupyter: necessary to run the notebooks, but nothing else

blas-devel
fftw

these two could be installed using a system package manager, but maybe we want to force pyshtools to use these packages anyways?

Reply:
pyproject.toml "dependencies" is exclusively for runtime pip-installable python library dependencies.

Meson-python is a build time dependency, not a runtime one.

Pkg-config (and development headers) are build time dependencies but also not python dependencies: they must be installed by the system (an OS or conda package manager).

There's an in-progress PEP to define a brand new way to specify external system dependencies: https://peps.python.org/pep-0725/

This, if/when accepted, will be the right place to list things like BLAS headers.